src/sys/unix/uds/datagram.rs

```diff
@@ -129,6 +129,29 @@ impl UnixDatagram {
     pub(crate) fn shutdown(&self, how: Shutdown) -> io::Result<()> {
         self.inner.shutdown(how)
     }
+
+    pub(crate) fn unlink(mut self) {
+        // Retrieve the local bound and drop the socket.
+        let maybe_path = {
```

Comment:
Nit: I''m not a fan of the name `maybe_path`. It's `maybe_path` because it's an `Option<Path>`, but the naming of `Option<T>` types don't usually follow a `maybe_...` naming convention. How about just `path` instead?